packages/gas-fee-controller/src/GasFeeController.ts

```diff
@@ -501,6 +479,13 @@ export class GasFeeController extends PollingController<
         state.estimatedGasFeeTimeBounds =
           gasFeeCalculations.estimatedGasFeeTimeBounds;
         state.gasEstimateType = gasFeeCalculations.gasEstimateType;
+        state.gasFeeEstimatesByChainId = state.gasFeeEstimatesByChainId || {};
+        state.gasFeeEstimatesByChainId[toHex(decimalChainId)] = {
+          gasFeeEstimates: gasFeeCalculations.gasFeeEstimates,
+          estimatedGasFeeTimeBounds:
+            gasFeeCalculations.estimatedGasFeeTimeBounds,
+          gasEstimateType: gasFeeCalculations.gasEstimateType,
+        } as any;
```

Comment:
This should give us better intellisense + warnings.

```suggestion
        } as SingleChainGasFeeState;
```

Reply:
thanks! done here: https://github.com/MetaMask/core/pull/1891/commits/1aca3181175ecdd6b79796a57271bd0a0ccca0ff